### ISSUE TITLE
AR ~4.2 counter_cache broken in 1.0.6

### ISSUE THREAD

**Issue by joel** [khiav223577/left_joins#13]
```
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "rails", "~> 4"
  gem "pg", "< 1"
  gem "left_joins", "1.0.6"
  gem 'sqlite3', '~> 1.3.6'
end

require "active_record"
require "minitest/autorun"
require "logger"
require "left_joins"

require 'sqlite3'

ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  self.verbose = true

  create_table :organizations, force: true do |t|
    t.string  :name
    t.integer :memberships_count, default: 0
    t.timestamps
  end

  create_table :memberships, force: true do |t|
    t.string  :name
    t.references :organization
    t.timestamps
  end
end

class Organization < ActiveRecord::Base
  has_many :memberships, dependent: :destroy
end 

class Membership < ActiveRecord::Base
  belongs_to :organization, counter_cache: :memberships_count
end

class BugTest < Minitest::Test
  def test_left_joins
    organization = Organization.create name: 'Organization 01'
    membership   = Membership.create name: 'A Memberships', organization: organization

    assert_equal organization.memberships.take, membership
  end
end
```

SQlite3

```
#<ActiveRecord::StatementInvalid: SQLite3::RangeException: column index out of range: UPDATE "organiz...ions" SET "memberships_count" = COALESCE("memberships_count", 0) + 1 WHERE "organizations"."id" = ?> with backtrace:
```

Postgres 

```
ActiveRecord::StatementInvalid:
       PG::ProtocolViolation: ERROR:  bind message supplies 2 parameters, but prepared statement "a5" requires 1
       : UPDATE "im_users" SET "memberships_count" = COALESCE("memberships_count", 0) + 1 WHERE "im_users"."id" = $1
```

**Comment by khiav223577** [comment]
Hi @joel ,

Thanks for your reporting.
I submitted a PR #14 to fix it :).

**Comment by khiav223577** [comment]
I just released 1.0.7.
